Document the new behavior of spacewalk-report (#1548)
uyuni-project/uyuni-docs: modules/administration/pages/reports.adoc

[[reports]]
= Generate Reports

{productname} allows the user to produce a variety of reports.
These reports can be helpful for taking inventory of your subscribed systems, users, and organizations.
Using reports is often simpler than gathering information manually from the {productname} {webui}, especially if you have many systems under management.

While the command line tool [command]``spacewalk-report`` can be used to generate pre-configured reports, with the introduction of the xref:specialized-guides:large-deployments/hub-reporting.adoc[Hub Reporting] it is also possible to generate fully customized reports.
This can be achieved by connecting any reporting tool that supports the SQL language to the reporting database and extract the data directly.

> For more information about the data availability and structure, please refer to the reporting database schema documentation.

Reviewer: ```suggestion
For more information about the data availability and structure, see the reporting database schema documentation.
```

== Using ``spacewalk-report``

To generate reports, you must have the [package]``spacewalk-reports`` package installed.

> 

Reviewer: ```suggestion
```
Drive-by edit. Avoid single sentence paragraphs in a row.

spacewalk-report <report_name>
----

[[spacewalk-report-reporting-db]]
== ``spacewalk-report`` and the reporting database

> [command]``spacewalk-report`` uses by default the new reporting database to extract the data.
> 
> This means that the new generated reports will have some differences in the structure and the format of the data.
> 
> The differences common to all reports are:

Reviewer: ```suggestion
[command]``spacewalk-report`` uses by default the new reporting database to extract the data.
This means that the new generated reports will have some differences in the structure and the format of the data.
The differences common to all reports are:
```
Avoid single sentence paragraphs in a row.

- the report data is not changing in real-time, but it's updated only by the execution of a scheduled task;

> - data duplication has been removed and columns that were previously considered "multival" contains now multiple values separated by ``;``. This also means that the command line options ``--multival-on-rows`` and ``--multival-separator`` are no longer applicable to the new reports, as their behavior is now the default;

Reviewer: ```suggestion
- data duplication has been removed and columns that were previously considered "multival" contain now multiple values separated by ``;``. This also means that the command line options ``--multival-on-rows`` and ``--multival-separator`` are no longer applicable to the new reports, as their behavior is now the default;
```

> - in all reports the new columns ``mgm_id`` and ``synced_date`` have been introduced to identify respectively the management server in the hub scenario and the last time the information was updated from the application database;

Reviewer: ```suggestion
- in all reports the new columns ``mgm_id`` and ``synced_date`` have been introduced to identify the management server in the hub scenario and the last time the information was updated from the application database;
```
I could be wrong but I think "respectively" is the wrong word here.  Do you mean "either ... or" or both at the same time?

- all boolean values are now represented by the ``True``/``False`` and not by a ``1``/``0`` values;

> - the column ``org_id`` has been replaced by ``organization``, which contains the organization name and not the numerical identifier.

Reviewer: ```suggestion
- the column ``org_id`` has been replaced by ``organization``, which contains the organization name and not the numerical identifier;
```
for consistency

- The term "server" has been replaced by "system". So, for example, the column ``server_id`` is now called ``system_id``.

> For report specific changes, please refer to the section <<report-list>>

Reviewer: ```suggestion
For report specific changes, see <<report-list>>.
```

[IMPORTANT]
====

> If these differences are problematic, the new option ``--legacy-report`` can be used to fall back to the old report which is executed against the application database.

Reviewer: ```suggestion
If this changed behavior causes trouble, the new option ``--legacy-report`` can be used to fall back to the old report, which is executed against the application database.
```
I'm not sure whether I got the meaning right.

Author: Yes, that's better :+1:
